### ISSUE TITLE
Switch off Note Metadata has some bugs

### ISSUE THREAD

**Issue by derwok** [GitJournal/GitJournal#162]
First: thanks for this cool app! Really like it!
Unfortunately currently my workflow is broken - see below:

I have to switch off creation of meta data, as I'm using other Markdown editors on desktop that can't handle YAML meta data (https://obsidian.md).

So I went to `Settings / Note Metadata Settings / Enable YAML header` and switched it off.

## **Four bugs** I encountered afterwards

1. If I now create new notes, I dont see the **Title field** while editing the note. I just can edit the body of the note. So its impossible to give the file a proper name.
1. In the Folder overview the new created notes are invisible, seems because they have no metadata? I can open the notes by clicking the white empty area. But its impossible to blindly say which file is which
1. The notes get a strange automatic date/time filename which I don't like. 
1. The file created in the repo looks like `2020-06-06-11-54-57` - without any file extension. The filename should be `.md` - the problem is: Obsidian markdown editor hides files without extension. => files should have a `.md`extension - even if the filename is auto generated

Hint:
I know I can rename a file. But thats cumbersome.

**Comment by vHanda** [comment]
Hey Wolfram.

Yes, there is a similar issue with `.txt` files.

1. Would #112 be a good enough alternative? That way you could do '# title' or GitJournal can even try to guess the title based on the first line?
2. Yup. This is bad. I'm currently working on reading the note metadata (created/modification date) from git instead. This way we will no longer need to store it in the file.
3. When there isn't a title available, what would you prefer?
4. Confirmed. Working on it right now.

**Comment by vHanda** [comment]
Fixed (4) with https://github.com/GitJournal/GitJournal/commit/e6be9aa228d40f59db4570ab8f8aaf3e5931f24d 

Making a new release right now.

**Comment by derwok** [comment]
Hey that's fast fixing. Thank you!

Regarding your question towards 1.)
I would still keep the "Title" entry field active even if Meta data is switched of.
Why? Title is very close to filename. And this is how you treat it currently when meta data is on. So - be it on or off - I would like to have that title field to be able to control the filename. Regarding #112 - I feel filename is not equal to heading#1. Almost any note I have, the filename gives the overal topic. (e.g. "Ardour" or "Obsidian" or "C++", ...) and the heading#1 is something different like "Installation", or "Basic Truth" or "Introduction". Or if I attend the meeting, then title/filename is the date & place & topic of the meeting while heading#1 is the first speaker or first disscused topic...
But I see one drawback with my approach: When I print such a note, the title is usually not printed - so when this info is nowhere else than in the UI, it is lost on print.
Nevertheless: I would vote for: `TITLE == FILENAME != HEADING#1`

Regarding question 2.) 
I think when I have the chance to specify a title in the UI (like when Metadatsa is switched on), everything is fine. I am in control of filename then. When thats not possible, a date stamp is OK. But the ".md" fix is important

Do you know when your fix will be available in the AppStore?

**Comment by vHanda** [comment]
How about when metadata is turned off, I convert the 'title' field to show the filename instead?

Regarding a new release, I'm trying to make a new release right now. So maybe today or tomorrow (Apple might take some time reviewing it).

**Comment by derwok** [comment]
Yep, that could work...
But why treat it different than with Metadata on?
Danke is see: if you show the full filename with extension, then users might accidently delete the precious file extension. If Metadata is off, you show the filename, but strip the .md/.txt extension?
And of course: When title changed, rename the file?

**Comment by jumble** [comment]
Hi all, happy to create a separate issue if you would prefer, but it seems that discussion on this issue has become related to the request I have. Essentially what I am interested in is the old title of this issue, "Allow Note Title to reflect the Filename". Like @derwok I also have been trying out Obsidian as an editor. My workflow currently involves using filenames as note titles. Adding an extra header containing the title isn't a major issue but is a mild inconvenience- at best it's unnecessary as I already know the title from the filename, and at worst whenever I update the note filename I need to ensure that the title header is in sync with it.

TLDR Would it be possible to add an option which makes the title field and note filenames one and the same?

**Comment by vHanda** [comment]
@jumble : Yes. I tried once. But supporting so many different configurations got in the way. I've added this to the list of tasks for this month. I'm trying to just keep a smaller number of tasks each month which I'm confident I can deliver.

Tldr: will do.
